protocols/kad/src/query/peers/fixed.rs

```diff
@@ -57,7 +57,12 @@ enum PeerState {
 }
 
 impl FixedPeersIter {
-    pub fn new(peers: Vec<PeerId>, parallelism: usize) -> Self {
+    pub fn new<I>(peers: I, parallelism: usize) -> Self
+    where
+        I: IntoIterator<Item = PeerId>
+    {
+        let peers = peers.into_iter().collect::<Vec<_>>();
```

Comment:
Happy for alternative suggestions. Using dynamic dispatching requires us to set `Send` and `Sync` bounds.

Doing the `collect` here instead of within `query.rs` moves the *hack* closer to its usage.

This might be optimized away by the compiler as well.